### ISSUE TITLE
Attention mask is None when 0.0 is not in attention_mask

### ISSUE THREAD

**Issue by mxjmtxrm** [huggingface/transformers#31271]
### System Info

- `transformers` version: 4.41.0.dev0
- Platform: Linux-5.15.0-88-generic-x86_64-with-glibc2.35
- Python version: 3.10.12
- Huggingface_hub version: 0.23.1
- Safetensors version: 0.4.3
- Accelerate version: 0.30.1
- Accelerate config: 	not found
- PyTorch version (GPU?): 2.3.0a0+ebedce2 (True)
- Tensorflow version (GPU?): not installed (NA)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: <fill in>
- Using distributed or parallel set-up in script?: <fill in>

### Who can help?

@ArthurZucker and @younesbelkada

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

In modeling_llama.py,  attention mask is None when 0.0 is not in attention_mask  with flash_attention:
```
        if self.config._attn_implementation == "flash_attention_2":
            if attention_mask is not None and 0.0 in attention_mask:
                return attention_mask
            return None
```
Why this? If I can modify `if attention_mask is not None and 0.0 in attention_mask:` to `if attention_mask is not None:`?

### Expected behavior

--

**Comment by ArthurZucker** [comment]
Hey! The reason is that we want to automatically dispatch to the optimised kernel of FA2 . When the attention mask is full of 1s, it means that there is no attention mask. Thus we don't send it to FA2

**Comment by mxjmtxrm** [comment]
Got it. Thanks
